draft-pauly-taps-arch.md

```diff
@@ -205,103 +205,93 @@ Beyond the basic objects, there are several high-level groups of actions that an
 
->> A Listener is any object that can be used to prepare a Connection with a remote endpoint without initiation by the local application. That is, it can passively accept initiations for Connections made by another endpoint.
+* Listener: A Listener object accepts incoming transport protocol connections from Remote Endpoints  and generates corresponding Connection objects. It is created from a Preconnection object that specifies the type of incoming connections it will accept.
 
 ### Pre-Establishment {#preestablishment}
 
-Endpoint
-
->> An Endpoint represents one side of a transport connection. It is a concept that represents how an application views itself or a peer that may vary in levels of specificity. Examples include "IP address + port", "hostname + port", and "DNS service name".
-
-Remote Endpoint
-
->> The Remote Endpoint in a properties \[MICHAEL: I can't parse "in a properties" here. Should this be removed?] represents the application's view of a peer that can participate in a transport connection.
-
-Local Endpoint
-
->> The Local Endpoint in a properties \[MICHAEL: I can't parse "in a properties" here. Should this be removed?] represents the application's view of itself that it wants to use for transport connections. This may be optional, or may be generic (a wild card address, for example).
-
-Path Selection Properties
-
->> Path Selection Properties consist of the options that an application may set to influence the selection of path between itself and the Remote Endpoint. These options can come in the form of requirements, prohibitions, or preferences. Examples of options which may influence path selection include the interface type (such as a Wi-Fi Ethernet connection, or a Cellular LTE connection), characteristics of the path that are locally known like Maximum Transmission Unit (MTU), or expected throughput or latency.
-
-Protocol Selection Properties
-
->> Protocol Selection Properties consist of the options that an application may set to influence the selection of transport protocol, configure the behavior of generic transport protocol features. These options come in the form of requirements, prohibitions, and preferences. Examples include reliability, service class, multipath support, and fast open support. 
+* Endpoint: An Endpoint represents one side of a transport connection. It is a concept that represents how an application views itself or a peer that may vary in levels of specificity. Examples include "IP address + port", "hostname + port", and "DNS service name".
 
-Specific Protocol Properties
+* Remote Endpoint: The Remote Endpoint in a properties \[MICHAEL: I can't parse "in a properties" here. Should this be removed?] represents the application's view of a peer that can participate in a transport connection.
```

Comment:
Would ICE create new endpoints or add additional information to an existing endpoint? In both cases I think this definition is to limited by only relating a remote to the application's view...

Reply:
For ICE I think we need: 1) `fn expand_candidates(p : Preconnection) -> List<Preconnection>` which can take a partially specified preconnection and expand it to a list of all possible candidates that match, so that list can be passed to the peer by the application; and 2) `fn initiate(peer_candidates : List<Preconnection>) ` which allows you to specify the remote candidates. I'll write this up, but it won't be today.